File: Java/1_Creational Patterns/2_Factory Method/Factory_Method.java
public abstract class Car {
    public abstract void accelerate();
    public abstract void brake();
    public abstract void steer();
}

public class SportsCar extends Car {
    public void accelerate() {
        // Implementation for sports car acceleration
    }

    public void brake() {
        // Implementation for sports car braking
    }

    public void steer() {
        // Implementation for sports car steering
    }
}

public class Sedan extends Car {
    public void accelerate() {
        // Implementation for sedan acceleration
    }

    public void brake() {
        // Implementation for sedan braking
    }

    public void steer() {
        // Implementation for sedan steering
    }
}

public class SUV extends Car {
    public void accelerate() {
        // Implementation for SUV acceleration
    }

    public void brake() {
        // Implementation for SUV braking
    }

    public void steer() {
        // Implementation for SUV steering
    }
}

public interface CarFactory {
    Car createCar();
}

public class SportsCarFactory implements CarFactory {
    public Car createCar() {
        return new SportsCar();
    }
}

public class SedanFactory implements CarFactory {
    public Car createCar() {
        return new Sedan();
    }
}

public class SUVFactory implements CarFactory {
    public Car createCar() {
        return new SUV();
    }
}

public class Main {
    public static void main(String[] args) {
        CarFactory factory = new SedanFactory();
        Car car = factory.createCar();
        car.accelerate();
        car.brake();
        car.steer();
    }
}

File: Python/1_Creational Patterns/2_Factory Method/Factory_Method.py
from abc import ABC, abstractmethod

class Car(ABC):
    @abstractmethod
    def accelerate(self):
        pass

    @abstractmethod
    def brake(self):
        pass

    @abstractmethod
    def steer(self):
        pass

class SportsCar(Car):
    def accelerate(self):
        # Implementation for sports car acceleration
        pass

    def brake(self):
        # Implementation for sports car braking
        pass

    def steer(self):
        # Implementation for sports car steering
        pass

class Sedan(Car):
    def accelerate(self):
        # Implementation for sedan acceleration
        pass

    def brake(self):
        # Implementation for sedan braking
        pass

    def steer(self):
        # Implementation for sedan steering
        pass

class SUV(Car):
    def accelerate(self):
        # Implementation for SUV acceleration
        pass

    def brake(self):
        # Implementation for SUV braking
        pass

    def steer(self):
        # Implementation for SUV steering
        pass

class CarFactory(ABC):
    @abstractmethod
    def create_car(self):
        pass

class SportsCarFactory(CarFactory):
    def create_car(self):
        return SportsCar()

class SedanFactory(CarFactory):
    def create_car(self):
        return Sedan()

class SUVFactory(CarFactory):
    def create_car(self):
        return SUV()

factory = SedanFactory()
car = factory.create_car()
car.accelerate()
car.brake()
car.steer()
